airflow/models.py

```diff
@@ -816,7 +816,8 @@ def pool_full(self, session):
             .first()
         )
         if not pool:
-            return False
+            raise ValueError('Task specified a pool ({}) but the pool '
```

Comment:
nit: use double quotes for sentences so that you don't have to escape apostrophes